com.unity.render-pipelines.universal/Editor/2D/Converter/BuiltInToURP2DMaterialUpgrader.cs

```diff
@@ -0,0 +1,136 @@
+using System;
+using System.IO;
+using System.Collections.Generic;
+using UnityEditor.Rendering.Universal.Converters;
+using UnityEngine;
+using UnityEngine.Rendering.Universal;
+
+
+namespace UnityEditor.Rendering.Universal
+{
+    internal sealed class BuiltInToURP2DMaterialUpgrader : RenderPipelineConverter
+    {
+        public override string name => "Material and Material Reference Upgrade";
+        public override string info => "This will upgrade all materials and material references.";
+        public override int priority => - 1000;
+        public override Type container => typeof(BuiltInToURP2DConverterContainer);
+
+        List<string> m_AssetsToConvert = new List<string>();
+
+        Material m_SpriteLitDefaultMat;
+        Material m_SpritesDefaultMat;
+        Material m_SpritesMaskMat;
+        Material m_SpriteMaskDefaultMat;
+        Shader   m_SpriteLitDefaultShader;
+        Shader   m_SpritesDefaultShader;
+
+        string m_SpritesDefaultShaderId;
+        string m_SpritesDefaultMatId;
+        string m_SpritesMaskMatId;
+
+        void UpgradeGameObject(GameObject go)
+        {
+            Renderer[] renderers = go.GetComponentsInChildren<Renderer>();
+            foreach (Renderer renderer in renderers)
+            {
+                if (!PrefabUtility.IsPartOfPrefabInstance(renderer))
+                {
+                    int materialCount = renderer.sharedMaterials.Length;
+                    Material[] newMaterials = new Material[materialCount];
+                    bool updateMaterials = false;
+
+                    for (int matIndex = 0; matIndex < materialCount; matIndex++)
+                    {
+                        if (renderer.sharedMaterials[matIndex] == m_SpritesDefaultMat)
+                        {
+                            newMaterials[matIndex] = m_SpriteLitDefaultMat;
+                            updateMaterials = true;
+                        }
+                        else if (renderer.sharedMaterials[matIndex] == m_SpritesMaskMat)
+                        {
+                            newMaterials[matIndex] = m_SpriteMaskDefaultMat;
+                            updateMaterials = true;
+                        }
+                        else
+                            newMaterials[matIndex] = renderer.sharedMaterials[matIndex];
+                    }
+
+                    if (updateMaterials)
+                        renderer.sharedMaterials = newMaterials;
+                }
+            }
+        }
+
+        public override void OnInitialize(InitializeConverterContext context, Action calback)
```

Comment:
Action callback, (my typo from long time ago that still lingers... sorry for that)